ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp

```diff
@@ -454,62 +450,155 @@ namespace ngraph
+            void InterpolateEval<T>::linear_onnx_generic_func(const T* input_data, T* out)
             {
                 size_t input_rank = m_input_data_shape.size();
-                size_t num_of_axes = m_axes.size();
 
-                assert((input_rank == 2) || (input_rank == 4));
-                assert((num_of_axes == 2) || (num_of_axes == input_rank));
+                assert(input_rank > 1);
 
-                bool correct_axes = ((m_axes[0] == 0) && (m_axes[1] == 1)) ||
-                                    ((m_axes[0] == 2) && (m_axes[1] == 3));
+                size_t num_of_axes = m_axes.size();
+
+                bool correct_axes = ((input_rank == 2) && (num_of_axes == 2) && (m_axes[0] == 0) &&
+                                     (m_axes[1] == 1)) ||
+                                    ((input_rank == 3) && (num_of_axes == 3) && (m_axes[0] == 0) &&
+                                     (m_axes[1] == 1) && (m_axes[2] == 2));
 
-                if ((num_of_axes == 4) && (input_rank == 4))
+                if (input_rank >= 4)
                 {
-                    correct_axes = (m_axes[0] == 0) && (m_axes[1] == 1) && (m_axes[2] == 2) &&
-                                   (m_axes[3] == 3);
+                    std::vector<int64_t> all_axes;
+                    std::vector<int64_t> axes_without_batch_and_channels;
+                    all_axes.push_back(0);
+                    all_axes.push_back(1);
+                    for (int64_t i = 2; i < static_cast<int64_t>(input_rank); ++i)
+                    {
+                        all_axes.push_back(i);
+                        axes_without_batch_and_channels.push_back(i);
+                    }
+
+                    correct_axes = ((num_of_axes == input_rank) && (m_axes == all_axes)) ||
+                                   ((num_of_axes == input_rank - 2) &&
+                                    (m_axes == axes_without_batch_and_channels));
                 }
 
                 assert(correct_axes);
 
-                const auto info = helper.get_info_for_linear_onnx_mode();
+                const auto info = helper.get_info_for_generic_linear_onnx();
 
                 int64_t batch_size = info.batch_size;
                 int64_t num_channels = info.num_channels;
-                int64_t output_height = info.output_height;
-                int64_t output_width = info.output_width;
-                int64_t input_height = info.input_height;
-                int64_t input_width = info.input_width;
+
+                auto& input_index_multipliers = info.input_index_multipliers;
+                auto& output_index_multipliers = info.output_index_multipliers;
+
+                int64_t input_data_ptr_increment = info.input_data_ptr_increment;
+                int64_t output_data_ptr_increment = info.output_data_ptr_increment;
+
+                auto& input_spatial_shape = info.input_spatial_shape;
+
+                int64_t axis_idx_offset = (input_rank == num_of_axes) * 2;
```

Comment:
Please explain the purpose of this code. You are multiplying the bool (`input_rank == num_of_axes``) to 2.

Reply:
Added comment about this variable, and this multiplication was rewritten using the ternary operator.